src/containers/App/App.js

```diff
@@ -21,40 +22,91 @@ import {
 } from 'react-router-dom';
 
 import {
+  Extension,
+  Extensions,
   Home,
   PipelineRun,
   PipelineRuns,
   Pipelines,
   Tasks,
   TaskRuns
 } from '..';
+import { fetchExtensions } from '../../actions/extensions';
+import { getExtensions } from '../../reducers';
 
 import '../../components/App/App.scss';
 
-const App = () => (
-  <Router>
-    <Switch>
-      <Route path="/" exact component={Home} />
-      <Redirect
-        from="/pipelines/:pipelineName"
-        exact
-        to="/pipelines/:pipelineName/runs"
-      />
-      <Redirect from="/tasks/:taskName" exact to="/tasks/:taskName/runs" />
-      <Route path="/pipelines" exact component={Pipelines} />
-      <Route path="/tasks" exact component={Tasks} />
-      <Route
-        path="/pipelines/:pipelineName/runs"
-        exact
-        component={PipelineRuns}
-      />
-      <Route path="/tasks/:taskName/runs" exact component={TaskRuns} />
-      <Route
-        path="/pipelines/:pipelineName/runs/:pipelineRunName"
-        component={PipelineRun}
-      />
-    </Switch>
-  </Router>
-);
+export class App extends Component {
+  componentDidMount() {
+    this.props.fetchExtensions();
+  }
+
+  render() {
+    const { extensions } = this.props;
+
+    return (
+      <Router>
+        <Switch>
+          <Route path="/" exact component={Home} />
+          <Redirect
+            from="/pipelines/:pipelineName"
+            exact
+            to="/pipelines/:pipelineName/runs"
+          />
+          <Redirect from="/tasks/:taskName" exact to="/tasks/:taskName/runs" />
+          <Route path="/pipelines" exact component={Pipelines} />
+          <Route path="/tasks" exact component={Tasks} />
+          <Route
+            path="/pipelines/:pipelineName/runs"
+            exact
+            component={PipelineRuns}
+          />
+          <Route path="/tasks/:taskName/runs" exact component={TaskRuns} />
+          <Route
+            path="/pipelines/:pipelineName/runs/:pipelineRunName"
+            component={PipelineRun}
+          />
+          <Route path="/extensions" exact component={Extensions} />
+          {extensions.map(({ displayName, name, source }) => {
+            if (!displayName) {
```

Comment:
What would cause this case ?

Reply:
The displayname and bundle-location annotations are optional, and only relevant for extensions that provide a UI. If they're omitted there's no need for us to support rendering a UI for that extension, so we skip it when outputting dynamic routes.